default.py

```diff
@@ -220,13 +220,37 @@ def PLANNED(payload, params):
 def PLANNED_PAGES(payload, params):
     return control.draw_items(_BROWSER.get_watchlist_planned(int(payload)), unbookmark_cm)
 
+@route('search_history')
+def SEARCH(payload, params):
+    history_delimiter=":_:"
+    history = control.getSetting("9anime.history")
+    history_array = history.split(history_delimiter)
+    if history != "" and "Yes" in control.getSetting('searchhistory') :
+        return control.draw_items(_BROWSER.search_history(history_array))
+    else :
+        return SEARCH(payload,params)
+
 @route('search')
 def SEARCH(payload, params):
+    history_delimiter=":_:"
+    history = control.getSetting("9anime.history")
+    history_array = history.split(history_delimiter)
     query = control.keyboard(control.lang(30007))
     if query:
+        if history != "" :
+            query = query+history_delimiter
+        history=query+history
+        while history.count(history_delimiter) > 6 :
+            history=history.rsplit(history_delimiter, 1)[0]
+        control.setSetting("9anime.history",history)
         return control.draw_items(_BROWSER.search_site(query))
-    return False
+    return False   
 
+@route('clear')
```

Comment:
where is the link for that?
havn't seen one..

Reply:
I was using a button to clean the history.
Hmm is there a function to go back to the main plugin window ?

Reply:
I'm putting back the clear button, it's in the bottom of the history list.